clients/unshaded/src/main/java/tachyon/client/file/FileOutStream.java

```diff
@@ -97,8 +100,8 @@ public FileOutStream(long fileId, OutStreamOptions options) throws IOException {
     }
     mClosed = false;
     mCanceled = false;
-    mHostname = options.getHostname();
     mShouldCacheCurrentBlock = mTachyonStorageType.isStore();
+    mLocationPolicy = FileWriteLocationPolicy.Factory.createPolicy(new TachyonConf(), options);
```

Comment:
If policy `CoolNewFileWritePolicy` wants options, we can put them into `CoolNewFileWriteLocationPolicyOptions`, which extends `FileWriteLocationPolicyOptions` and can therefore be set inside of `OutputStreamOptions`. This is nice because it avoids having various fields in `OutputStreamOptions` which only apply to one type of `FileWriteLocationPolicy`